lib/github/parse_owner.ex

```diff
@@ -56,14 +60,30 @@ defmodule BorsNG.CodeOwnerParser do
             end
           end)
       end)
-
+ 
     required_reviewers = Enum.filter(required_reviewers, fn x -> x != nil end)
 
     Logger.debug("Required reviewers: #{inspect(required_reviewers)}")
 
     required_reviewers
   end
 
+  @spec process_double_asterisk(String.t, String.t) :: boolean
+  def process_double_asterisk(file_name, file_pattern) do
+    double_asterisk = "**"
+    cond do
+      String.starts_with?(file_pattern, double_asterisk) ->
+        pattern = String.trim_leading(file_pattern, double_asterisk)
+        String.contains?(file_name, pattern)
+      String.ends_with?(file_name, double_asterisk) ->
+        pattern = String.trim_leading(file_pattern, double_asterisk)
```

Comment:
Wait, it's really checking the `file_name` for a "`**`"? I can't find anything about it in GitHub's documentation. It only mentions asterisks as being special in patterns.

Reply:
Accordingly to https://help.github.com/en/articles/about-code-owners: `A CODEOWNERS file uses a pattern that follows the same rules used in gitignore files.` https://git-scm.com/docs/gitignore#_pattern_format shows how the pattern support is for `**`.

Reply:
That being said, I did a very simple logic that I'll try to explain. For example, if the pattern is `abc/**` I remove the `**` asterisk and check if the file name starts with `abc/`. The other way around if it's `**/foo`, I remove the `**` and check if the file name has `foo` on it. Did I managed to explain correctly? Not super sure if it's the best approach. I'll be glad to change if you have a better idea.

Reply:
Yes, but that's for the *patterns*. `String.ends_with?(file_name, double_asterisk)` checks whether the file's own name ends with double asterisks.

Reply:
Oh, wait, but in this part I should be removing the trailing one...

Reply:
I added tests for all the cases described as a proof of concept

Reply:
I totally misunderstood you, I'm sorry, you are correct, just fixed it.